sdformat_rs/src/lib.rs

```diff
@@ -11,10 +12,186 @@ use yaserde_derive::{YaDeserialize, YaSerialize};
 // Most of the structs are generated automatically from the
 include!(concat!(env!("OUT_DIR"), "/sdf.rs"));
 
+#[derive(Default, PartialEq, Clone, Debug)]
+pub enum ElementData {
+    #[default]
+    Empty,
+    Integer(i64),
+    Double(f64),
+    String(String),
```

Comment:
TBH its possible to have `Vec3` and `Vec2` as inputs in SDF. The C++ way this in C++ is to use template substitution for get. My recommendation would be to store it as a string and then create and `impl` for `From`/`Into`.

Reply:
Yep... `TryFrom` works.

Reply:
https://github.com/open-rmf/sdf_rust_experimental/pull/5/commits/ec4d6638548bf68f63007d2ab7a23cd9f211d270. I couldn't really find `Vector2` types so I only did the `Vector3d` / `Vector3i` implementations

Reply:
Its in the XSD spec here: https://github.com/open-rmf/sdf_rust_experimental/blob/main/sdformat_rs/sdformat_spec/1.10/schema/types.xsd

Fair enough we also have pose, color, quaternion and time. This is quite reasonable for now in my opinion. We can open a ticket to support this later.